code/obj/item/flamethrower.dm

```diff
@@ -20,28 +30,130 @@ GETLINEEEEEEEEEEEEEEEEEEEEE
+	var/min_temperature = FLAMER_MIN_TEMP
+	/// Maximum temperature this thing'll set the reagents to
+	var/max_temperature = FLAMER_MAX_TEMP
+	/// Can this thing vary the temperature?
+	var/adjustable_temp = 1
+	/// How much chem to try to stuff into the projectile
+	var/amt_chem = FLAMER_DEFAULT_CHEM_AMT
+	/// Maximum volume this thing can shoot per shot
+	var/amt_chem_max = FLAMER_MAX_CHEM_AMT
+	/// Can this thing vary the amount of stuff loaded?
+	var/adjustable_chem_amt = 1
 	var/obj/item/tank/gastank = null
+	var/obj/item/fueltank = null // Honestly, anything with reagents'll do. Just needs sprites!
+	/// Can we swap out the tanks? Mainly so backpack flamers don't lose their flamer
+	var/swappable_tanks = 1
+	/// Is this a fancy combat flamethrower? Boosts melee damage, reduces spread
+	var/combat_flamer = 0
+	current_projectile = new/datum/projectile/special/shotchem
 	contraband = 5 //Heh
 	m_amt = 500
 	stamina_damage = 15
 	stamina_cost = 15
 	stamina_crit_chance = 1
 	move_triggered = 1
-
-	proc/get_reagsource(mob/user)
-		return null
+	/// Burstfire, but actually
+	use_shootloop = 1
+	refire_delay = 4 DECI SECONDS
+	burst_count = 3
+	spread_angle = 5
 
 	New()
 		..()
 		BLOCK_SETUP(BLOCK_LARGE)
 		setItemSpecial(null)
 
-/obj/item/flamethrower/assembled
+	/// Just check if there's a usable air and fuel tank
+	canshoot()
+		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
+			return TRUE
+
+	/// check for tank, pressure in tank, fuelltank, fuel in tank, and... then dump the stuff into it!
+	process_ammo(var/mob/user)
+		var/turf/T = get_turf(src)
+		var/datum/gas_mixture/T_env = T.return_air()
+		if(!src.fueltank)
+			boutput(user, "<span class='alert'>This [src] doesn't have a fuel source!</span>")
+			return FALSE
+		else if(!(src.fueltank in src.contents) && !(src.fueltank in user.get_equipped_items())) // Tank is loaded
+			boutput(user, "<span class='alert'>You need to either wear [src]'s fuel source or load it into the weapon!</span>")
+			return FALSE
+		else if(src.fueltank?.reagents.total_volume <= 0)
+			boutput(user, "<span class='alert'>This [src]'s fuel source is empty!</span>")
+			return FALSE
+		else if(T_env && src.gastank?.air_contents)
+			if(MIXTURE_PRESSURE(T_env) > MIXTURE_PRESSURE(gastank.air_contents))
+				boutput(user, "<span class='alert'>Not enough pressure in [src]'s gas tank to operate!</span>")
+				return FALSE
+		return TRUE
+
+	alter_projectile(var/obj/projectile/P)
+		if(!P.proj_data)
+			return
+
+		if(!canshoot())
+			return
+
+		var/obj/projectile/SD = P.special_data
+		var/datum/reagents/FR = src.fueltank.reagents
+		var/datum/gas_mixture/GTAC = src.gastank.air_contents
+
+		var/chem_amount = min(src.fueltank?.reagents.total_volume, src.amt_chem)
+		var/datum/reagents/chems = new(chem_amount)
+		FR.trans_to_direct(chems, chem_amount)
+		SD["crossed_turfs"] = list()
+		SD["FT_reagents"] = chems
```

Comment:
Its so I can have the reagents stored in the special_data react the chems inside the projectile itself to ensure a fireflash if there's going to be one. Not the best idea, but fireflashes started being more consistent after doing this. I'll find a better solution.

Reply:
Also turns out storing the reagents in a physical object exposes the reagents to any hotspots on the turf, which is a problem since the reagents stored in the projectile are supposed to be more of a template. Or the for loop is counting the projectile as part of the turf's contents and setting its temperature. Either way, gotta protect the chems. Probably exclude probjectiles from the loop.